File: src/backend/Domain/Models/PrintJob.cs
namespace Domain.Models;

public class PrintJob
{
    public Guid Id { get; set; }
    public Guid PrinterId { get; set; }
    public Guid? SpoolId { get; set; }
    public string? PrintFileName { get; set; }
    public string? TaskId { get; set; }
    public PrintJobStatus Status { get; set; }
    public float GramsUsed { get; set; } = 0;
    public DateTime StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }
    public DateTime LastUpdatedAt { get; set; }
    public string Source { get; set; } = "mqtt";
    public string? Notes { get; set; }
    public bool FilamentDeducted { get; set; } = false;
    public int? EstimatedFinishTime { get; set; }
    public Printer Printer { get; set; } = null!;
    public Spool? Spool { get; set; }
    public ICollection<PrintJobFilament> Filaments { get; set; } = [];
}

File: src/frontend/web/src/types/printJob.ts
export interface PrintJobFilamentResponse {
  id: string
  spoolId: string | null
  colorName: string | null
  colorHex: string | null
  material: string | null
  gramsUsed: number
  slotIndex: number
}

export interface PrintJobResponse {
  id: string
  printerId: string
  printerName: string | null
  spoolId: string | null
  spoolBrand: string | null
  spoolColorName: string | null
  spoolColorHex: string | null
  spoolMaterial: string | null
  printFileName: string | null
  taskId: string | null
  status: 'running' | 'paused' | 'finished' | 'failed' | 'cancelled' | 'unknown'
  gramsUsed: number
  filamentDeducted: boolean
  startedAt: string
  finishedAt: string | null
  estimatedFinishTime: number | null
  source: string
  notes: string | null
  filaments: PrintJobFilamentResponse[]
}

export interface PrintJobPagedResponse {
  total: number
  page: number
  limit: number
  jobs: PrintJobResponse[]
}
